table/block_based/index_builder.cc

```diff
@@ -162,7 +167,12 @@ void PartitionedIndexBuilder::AddIndexEntry(
                                       first_key_in_next_block, block_handle);
     sub_index_last_key_ = std::string(*last_key_in_current_block);
     if (sub_index_builder_->seperator_is_key_plus_seq_) {
-      // then we need to apply it to all sub-index builders
+      // then we need to apply it to all sub-index builders and reset
+      // flush_policy to point to Block Builder of sub_index_builder_ that store
+      // internal keys.
+      flush_policy_.reset(FlushBlockBySizePolicyFactory::NewFlushBlockPolicy(
```

Comment:
This is an allocation per index entry which might be nontrivial. I think the easy way to avoid these extra allocations is to reset the flush policy only when `!seperator_is_key_plus_seq_`.

Reply:
Initially I added the check but then number of partitions got changed (it was changed from 11 to 2). So I couldn't understand why the value got changed by adding additional check ```!seperator_is_key_plus_seq_```.

Reply:
I see you added it back. Does it behave as expected now (produce partitions around `metadata_block_size` bytes) or is there still an unexpectedly small number of partitions?

Reply:
Oh my bad. The check  at line 132 is fine so i added that but by adding check here (line 170) partitions are still 2 (I forgot to remove this check while uploading the diff). I am still not sure why this is the case. I have added logs to look further for debugging.

Reply:
discussed offline, seems it is producing the expected number of partitions now.